lib/Service/AiIntegrations/Cache.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @copyright 2017 Hamza Mahjoubi<hamzamahjoubi22@proton.met>
+ *
+ * @author 2017 Hamza Mahjoubi <hamzamahjoubi22@proton.me>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Mail\Service\AiIntegrations;
+
+use OCP\ICache;
+use OCP\ICacheFactory;
+
+class Cache {
+	// Cache for one week
+	public const CACHE_TTL = 7 * 24 * 60 * 60;
+
+	/** @var ICache */
+	private $cache;
+
+
+	public function __construct(ICacheFactory $cacheFactory) {
+		$this->cache = $cacheFactory->createDistributed('mail.ai');
```

Comment:
distributed caches are always more expensive than local ones because of the net connection they have to establish. local caches can use sockets.

let's use a local cache instance for this. in the worst case one thread is summarized a second time. so be it.